backend/coding-standards.md

```diff
@@ -24,6 +24,32 @@ For our PHP projects we use [PHPCS](https://github.com/squizlabs/PHP_CodeSniffer
 
 We also use [PHPCD](https://github.com/sebastianbergmann/phpcpd) to ensure that code is not duplicated within the project.
 
+## AWS Resource Naming
+
+Historic resources don't follow an agreed convention. However, for all new resources, both in old and new projects, we should follow the schema:
+
+`${service-name}-${service-stage}-${resource-type}-${resource-label}`
```

Comment:
I have only one concern, as AWS has limits on name length for pretty much everything, We may not be able to apply this rule everywhere.
I would opt for the shortest significant naming, so `${service-name}-${service-stage}` as a starter, then ${resource-label} would be added if needed.

Reply:
I disagree! I would advise to go towards the opposite route, with explicit, long names by default and only truncating where the resource goes over the maximum allowed by AWS (which is 80, isn't it?)

Even in lengthy service names we don't get close to 80, for example:
- `serverless-prize-platform-production-dynamodb-prize-categories` is 62 characters

Reply:
However, if we want to save characters, I would advise to change the `serverless` prefix into `sls`, which is a considerable saving

Reply:
Ok, won't argue with either route to take.
It's an extreme case, where the limitation will dictate the naming anyway, and unfortunately it's sometimes less than 80, 64 in some instances!
I would just prefer to avoid redundant info in name, like dynamodb name including `dynamodb`!

Reply:
I think we can skip the resource type in the name, it was there just to highlight what the resource is by just reading the name. The use case where it makes sense is correlated resources, such as

- `serverless-prize-platform-staging-s3-tickets`
- `serverless-prize-platform-staging-glue-tickets`

Or

- `supporter-event-service-staging-sqs-main`
- `supporter-event-service-staging-sqs-dlq-main`

I am not precious about the resource type, but I think the label is important, even if it is just `-main` (+5 characters)

Reply:
Was also wondering if char limits could be a concern, or even just UIs where it's hard to read long names.

However, `resource-label` seems like the most important part of the name to me and don't think it can be dropped. It's the key identifier isn't it? `sls-contact-store-staging-s3` is pretty useless as a name.

Still don't get the need for the resource type (s3, dynamodb) as I'm not aware of situations when we'd be scanning the names of resources without knowing what type we're looking at, although that's not to say there aren't any!

Reply:
Actually I take that back. I guess there probably will be lots of times we only want _one_ 'resource' e.g. S3 bucket for a given service.... :)

Reply:
It is also because I had the case where I named a S3 bucket and a Glue Table the same way and in error deploys I kept working on the table, totally forgot about the bucket, spent a lot of time with an unhelpful error from serverless (my fault for misleading naming).

See my example above. How would we differentiate a s3 bucket and the related glue table? They can't share the same name.

`service`, `stage` and `label` are the fundamental bits. If the resulting string is too long, the `service` bit is probably the bit we need to make shorter (for future services)

Reply:
> However, if we want to save characters, I would advise to change the serverless prefix into sls, which is a considerable saving

We could get rid of `serverless-` completely, since every service we deploy will be serverless... I think the GitHub repo name should still be `serverless-prize-platform`, but the service name as defined in `serverless.yml` could omit it.

Pro: saves 11 characters; avoids noise in names; some AWS filtering works on prefix only, so currently always have to always type "serverless".
Con: service name doesn't exactly match repo name, but that would happen anyway if we opted for `sls`, and GitHub's search is not prefix-based.